### PR TITLE
HDPath parsePath - fixed issue with parsing HDPaths for private keys

### PR DESCRIPTION
**Pull request by milczarekIT**
Fixed: https://github.com/bitcoinj/bitcoinj/issues/1992

### CODE REVIEW

**Comment by milczarekIT** [comment]
Up

File: core/src/main/java/org/bitcoinj/crypto/HDPath.java [hunk 1]
```diff
@@ -151,19 +151,22 @@ public static HDPath m(ChildNumber... children) {
      * Where a letter "H" means hardened key. Spaces are ignored.
      */
     public static HDPath parsePath(@Nonnull String path) {
-        String[] parsedNodes = path.replace("M", "").split("/");
+        boolean hasPrivateKey = path.startsWith(PREFIX_PRIVATE);
+        String[] parsedNodes = path.replace(PREFIX_PUBLIC, "")
+                .replace(PREFIX_PRIVATE, "")
+                .split("/");
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/crypto/HDPath.java:157]: Maybe use the SEPARATOR constant at this point?

File: core/src/main/java/org/bitcoinj/crypto/HDPath.java [hunk 2]
```diff
@@ -151,19 +151,22 @@ public static HDPath m(ChildNumber... children) {
      * Where a letter "H" means hardened key. Spaces are ignored.
      */
     public static HDPath parsePath(@Nonnull String path) {
-        String[] parsedNodes = path.replace("M", "").split("/");
+        boolean hasPrivateKey = path.startsWith(PREFIX_PRIVATE);
+        String[] parsedNodes = path.replace(PREFIX_PUBLIC, "")
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/crypto/HDPath.java:155]: Actually I think the whole .replace() is wrong. Only a leading `M` or `m` should be removed. Later occurences would be illegal and should lead to some form of error. (I'm aware that the previous implementation had this problem too.)

File: core/src/main/java/org/bitcoinj/crypto/HDPath.java [hunk 3]
```diff
@@ -151,19 +151,22 @@ public static HDPath m(ChildNumber... children) {
...
+                .replace(PREFIX_PRIVATE, "")
+                .split("/");
         List<ChildNumber> nodes = new ArrayList<>();
 
         for (String n : parsedNodes) {
-            n = n.replaceAll(" ", "");
+            n = n.replace(" ", "");
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/crypto/HDPath.java:161]: This change seems unrelated to me.

File: core/src/test/java/org/bitcoinj/crypto/HDPathTest.java [hunk 4]
```diff
@@ -31,7 +33,7 @@
     public void testPrimaryConstructor() throws Exception {
         HDPath path = new HDPath(true, Collections.<ChildNumber>emptyList());
         Assert.assertTrue("Has private key returns false incorrectly", path.hasPrivateKey);
-        Assert.assertEquals("Path not empty", path.size(), 0);
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/crypto/HDPathTest.java:34]: Fixing the use of `assertEquals()` in the entire project would be a nice separate PR.

**Comment by schildbach** [comment]
I fixed this via ed845fd22b58ffb7de0b50a1f2e22f34b5d73277
